home-feature/src/main/java/com/ttp/feature/home/ui/CarouselPlayerView.kt

```diff
@@ -0,0 +1,65 @@
+package com.ttp.feature.home.ui
+
+import android.content.Context
+import android.util.AttributeSet
+import android.widget.FrameLayout
+import androidx.core.view.isVisible
+import com.google.android.exoplayer2.ExoPlayer
+import com.google.android.exoplayer2.ui.PlayerView
+import com.ttp.core.coroutines.viewScope
+import com.ttp.entities.Game
+import com.ttp.extensions.android.inflate
+import com.ttp.feature.home.R
+import com.ttp.feature.home.extensions.play
+import com.ttp.feature.home.extensions.toMediaSource
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.map
+
+private const val PLAYBACK_DELAY_TIME = 2000L
+
+internal class CarouselPlayerView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : FrameLayout(context, attrs, defStyleAttr) {
+
+    private val playerVew: PlayerView
+
+    init {
+        isVisible = false
+        inflate(R.layout.view_carousel_player, true)
+
+        playerVew = findViewById(R.id.carousel_player)
+        playerVew.clipToOutline = true
+    }
+
+    fun bind(data: Game, player: ExoPlayer, playerController: Flow<Long>) {
+
+        viewScope.launchWhenAttached {
```

Comment:
`viewScope.launchWhenAttached` può essere invocato negli adapter solo nella `onViewAttachedToWindow` se vuoi ascoltare eventi. Questo perché la `attach` e la `detach` potenzialmente possono essere chiamate back and forth senza un `bind`. Ti consiglio questo video https://www.youtube.com/watch?v=KhLVD6iiZQs&feature=youtu.be&t=1528 di Yigit che spiega un po' il lifecycle degli adapter

Reply:
🇮🇹